controllers/controllers/resource/fetcher.go

```diff
@@ -641,8 +643,16 @@ func MapMachineTemplateToCloudStackMachineConfigSpec(csMachineTemplate *cloudsta
 	if csSpec.Spec.Symlinks == nil {
 		csSpec.Spec.Symlinks = map[string]string{}
 	}
-	for key, element := range csMachineTemplate.Spec.Spec.Spec.Symlinks {
-		csSpec.Spec.Symlinks[key] = element
+	for _, keyValueStr := range strings.Split(csMachineTemplate.Annotations["symlinks."+cloudstackAnnotationSuffix], ",") {
+		keyValueStr = strings.TrimSpace(keyValueStr)
+		if len(keyValueStr) == 0 {
+			continue
+		}
+		keyV := strings.Split(keyValueStr, ":")
+		if len(keyV) != 2 {
+			return nil, fmt.Errorf("symlinks: %s is not key:value format", keyV)
+		}
+		csSpec.Spec.Symlinks[keyV[0]] = keyV[1]
```

Comment:
could you move this to a helper function?

Reply:
create a helper method, but still need to keep 647 - 650 here due to `continue` interrupt exec flow.